examples/infection-by-group/README.md

```diff
@@ -0,0 +1,380 @@
+
+As in other `ixa` models, the simulation is managed by a central `Context` object which loads parameters, initializes user-defined modules, and starts a callback execution loop:
+
+```rust
+struct Parameters {
+    random_seed: u64,
+    population_size: usize,
+    k: usize,
+    foi: HashMap<Counties, f64>,
+    base_infection_period: f64,
+    movement: HashMap<HashMap<Counties, f64>>
+}
+
+let context = Context::new();
+context::load_parameters<Parameters>("config.toml")
+
+// Initialize modules
+mod people
+mod transmission
+mod infection
+mod regions
+
+people.init();
+transmission.init();
+infection.init();
+regions.init();
+
+// Run the simulation
+context.execute();
+```
+
+### People and person properties
+
+People must be added to the simulation and have some properties assigned to them, such as their infection status. Their infection time is relevant for scheduling their recovery (more on this below). In this example, I tether assigning people to groups to moving them between regions, and do the assignment there. However, this is open to discussion.
+
+In `people.rs`:
+
+```rust
+enum Properties {infection_status, infection_time}
+// infection_time is Option<f64> with default value None
+enum InfectionStatus {S, I, R}
+
+fn init(context: &mut Context) {
+
+    // create the population and make them all susceptible -- ignoring their region for now
+    for id in 0..population_size {
+        let Person person = context.add_person();
+        person.set_property_value(Properties::infection_status, InfectionStatus::S);
+    }
+}
+```
+
+### Movement between regions
+
+In `regions.rs`:
+
+```rust
+use strum_macros::EnumIter;
+use ixa::GroupMappings::*;
+
+enum GroupTypes {Counties}
+
+#[derive(Debug, EnumIter)]
+enum Counties {A, B}
+
+define_rng!(MovementRng);
+define_group!(GroupType::Counties,
+              Counties,
+              GroupMappings::one_to_one);
+
+fn init(context: &mut Context) {
+    // assign people to regions and plan their movement upon their creation
+    // this may not be the structure we ultimately decide upon for assigning
+    // information to people upon their creation
+    panic!(parameters.get_parameter(k) < parameters.get_paramter(population_size));
+    context.subscribe_to_event(PersonCreationEvent, assign_and_plan_movement);
+}
+
+// note there is an alternate implementation where the per-region movement rate is calculated,
+// and then a random person is selected within the region to move.
+fn assign_and_plan_movement(context: &mut Context,
+                            person_creation_event: PersonCreationEvent) {
+    // evenly distribute ppl among the counties
+    let county_index = parameters.get_parameter(population_size) % parameters.get_parameter(k);
```

Comment:
Is there a reason you want to do this differently from moving them (i.e. iterate through instead of randomly distributing them on the initial assignment?)

I would imagine this would end up as either an `assign_random_group` function on `PersonBuilder` (if this is how we end up interacting with person properties) or this is a `assign_random_group` is a trait extension on context that takes a person id, and you set up the listener here.

Reply:
In GCM, we could only make subscription functions that took context and the event as an argument, and nothing else! So, we had to use this kind of pattern a lot. We have changed this, so updating! Thanks!

Reply:
Oh, I think I properly understand your point here. This construction ensures that the two populations start off exactly equal, rather than on average equal.